Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -744,6 +744,17 @@ extension NSManagedObjectContext: MessageDatabaseSession {
 
     func message(id: MessageId) -> MessageDTO? { .load(id: id, context: self) }
 
+    func messageExists(id: MessageId) -> Bool {
+        let request = NSFetchRequest<MessageDTO>(entityName: MessageDTO.entityName)
+        request.predicate = NSPredicate(format: "id == %@", id)
+        do {
+            let count = try count(for: request)
+            return count != 0
```

Comment:
Still to be tested whether this approach using `count` is better than a batch `fetch` using all the ids.
Theory says that individual `count` should be better 🤷

Reply:
do you have some resource to share about the performance comparison between `count` and `fetch`?

Reply:
Added to the description